src/rdkafka_fetcher.c

```diff
@@ -616,6 +742,10 @@ rd_kafka_fetch_reply_handle(rd_kafka_broker_t *rkb,
         const int log_decode_errors = LOG_ERR;
         rd_kafka_topic_t *rkt       = NULL;
         int16_t ErrorCode           = RD_KAFKA_RESP_ERR_NO_ERROR;
+        uint64_t _tagcnt;
+        uint64_t _tag, _taglen;
+        rd_kafkap_Fetch_reply_tags_t fetch_tags = {0};
+        rd_bool_t leader_tags_present           = rd_false;
```

Comment:
We can have a boolean that is true if this RPC version supports tags, only in that case we allocate the topic array.
In the tags there's the field that will have the same scope of `leader_tags_present` that is `topics_with_leader_change_cnt` that is used to allocate in the tmpabuf only the topics (and then the partitions) that have leader changes.
```suggestion
        rd_bool_t has_fetch_tags               = rd_false;
```

Reply:
Updated